pkg/controller/resourceclaim/controller.go

```diff
@@ -690,11 +689,10 @@ func (ec *Controller) findPodResourceClaim(pod *v1.Pod, podClaim v1.PodResourceC
 		}
 
 		// No annotation? It might a ResourceClaim created for
-		// the pod with a previous Kubernetes release where the
-		// ResourceClaim name was deterministic, in which case
+		// the pod with a previous Kubernetes release, in which case
 		// we have to use it and update the new pod status
 		// field accordingly.
-		if !ok && claim.Name != deterministicName {
+		if !ok {
```

Comment:
/lgtm cancel

This still mentions "created for a previous Kubernetes". Here's how I would describe it:
```
// No annotation? Then it cannot be an automatically generated claim
// and we need to ignore it.
```

This `ok` check can be moved directly after `podClaimName, ok := claim.Annotations[podResourceClaimAnnotation]`.

The next if check then is 
```
// Not the claim for this particular pod claim?
if podClaimName != podClaim.Name {
			continue
		}
```

Reply:
updated.

Reply:
Can you also change the order of the if checks so that there is no duplication?

```
podClaimName, ok := claim.Annotations[podResourceClaimAnnotation]
if !ok {
...
}

if podClaimName != podClaim.Name {
...
}
```